### ISSUE TITLE
Throwable vs. Exception

### ISSUE THREAD

**Issue by PhoneBuster** [zendframework/zend-mvc#146]
I use zend-mvc 2.7.8. If i have a bad argument in my own factory the framework throws the following error :

**Fatal error: Uncaught TypeError: Argument 4 passed to Zend\Mvc\DispatchListener::marshalBadControllerEvent() must be an instance of Exception, instance of TypeError given...**

To fix this i've to change the the argument type in the DispatchListener class (method marshalBadControllerEvent) from 

``` php
   protected function marshalBadControllerEvent(
        $controllerName,
        MvcEvent $event,
        Application $application,
        \Exception $exception
    ) {
        $event->setName(MvcEvent::EVENT_DISPATCH_ERROR);
        $event->setError($application::ERROR_EXCEPTION);
        $event->setController($controllerName);
        $event->setParam('exception', $exception);

        $events  = $application->getEventManager();
        $results = $events->triggerEvent($event);
        $return  = $results->last();
        if (! $return) {
            return $event->getResult();
        }

        return $return;
    }
```

to

``` php
   protected function marshalBadControllerEvent(
        $controllerName,
        MvcEvent $event,
        Application $application,
        Throwable $exception
    ) {
        $event->setName(MvcEvent::EVENT_DISPATCH_ERROR);
        $event->setError($application::ERROR_EXCEPTION);
        $event->setController($controllerName);
        $event->setParam('exception', $exception);

        $events  = $application->getEventManager();
        $results = $events->triggerEvent($event);
        $return  = $results->last();
        if (! $return) {
            return $event->getResult();
        }

        return $return;
    }
```

**Comment by Ocramius** [comment]
Due to PHP 5.x compatibility, the quickest solution for now is to drop the type-hint completely

**Comment by PhoneBuster** [comment]
My PHP Version is 7

**Comment by Ocramius** [comment]
The library still supports some versions of PHP 5.x

Marco Pivetta

http://twitter.com/Ocramius

http://ocramius.github.com/

**Comment by weierophinney** [comment]
Fixed with #153 and release 2.7.9. You may also need to update your error templates per the suggestions in #157.

**Comment by andreaventuri** [comment]
I still have the same problem in version 3.0.1

**Comment by michalbundyra** [comment]
It was fixed only on v2.7, here you have issue for v3: https://github.com/zendframework/zend-mvc/pull/161

/cc @weierophinney

**Comment by weierophinney** [comment]
Fixed with #161.
